projects/js-packages/components/components/text/index.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.module.scss';
+import type { H3Props, TextProps, TitleProps } from './types';
+import { BOX_MODEL_VALUES, VARIANTS_MAPPING } from './constants';
+
+/**
+ * Text component.
+ *
+ * @param {TextProps} props - Component props.
+ * @returns {React.ReactElement} - JSX.Element
+ */
+const Text: React.FC< TextProps > = ( {
+	variant = 'body',
+	children,
+	component,
+	className,
+	...componentProps
+} ) => {
+	const Component = ( component || VARIANTS_MAPPING[ variant ] || 'span' ) as React.FC< TextProps >;
+
+	// Build Styles module CSS classnames.
+	const boxModelClasses = useMemo( () => {
+		return BOX_MODEL_VALUES.reduce( ( acc, value ) => {
+			if ( typeof componentProps[ value ] !== 'undefined' ) {
+				acc += styles[ `${ value }-${ componentProps[ value ] }` ] + ' ';
+				// pick spacing prop. Do not pass down to Component.
+				delete componentProps[ value ];
+			}
+			return acc;
+		}, '' );
+	}, [ componentProps ] );
+
+	return (
+		<Component
+			className={ classNames( styles.reset, styles[ variant ], className, boxModelClasses ) }
+			{ ...componentProps }
+		>
+			{ children }
+		</Component>
+	);
+};
+
+export default Text;
+
+/**
+ * Heading component - Medium size.
+ *
+ * @param {TextProps} props      - Component props.
+ * @returns {React.ReactElement} - JSX.Element
+ */
+export const H2: React.FC< TextProps > = ( { children, ...componentProps } ) => (
+	<Text variant="headline-medium" mb={ 3 } { ...componentProps }>
+		{ children }
+	</Text>
+);
+
+/**
+ * Heading component - Small size,
+ *
+ * @param {H3Props} props        - Component props.
+ * @returns {React.ReactElement} - JSX.Element
+ */
+export const H3: React.FC< H3Props > = ( { children, weight = 'bold', ...componentProps } ) => {
+	const normalizedWeight = ( weight === 'bold' ? '' : weight ) as string;
```

Comment:
Is it possible to reuse the param?
```suggestion
	weight = weight === 'bold' ? 'bold' : weight;
```

Reply:
<img width="642" alt="image" src="https://user-images.githubusercontent.com/77539/166959958-dfbf4f74-8903-40ee-8531-cb01e375ed44.png">

Reply:
Also, the existing behaviour makes that normalised weight to be empty when `bold` is passed and that's what the styles file also expects.

Reply:
But looking at the code we could improve it and remove this `normalizedWeight`.

```es6
const variant = `headline-small${
    weight === 'bold' ? '' : `-${ weight }`
}` as TextProps[ 'variant' ];
```

Reply:
Done ✅